wgpu/_coreutils.py

```diff
@@ -222,3 +224,47 @@ def help(*searches, dev=False):
         # Print lines
         if lines:
             print("\n".join("    " + line for line in lines))
+
+
+class _Formatter(logging.Formatter):
+    """Formatter that enabled PM debugging and shortens recurring error messages."""
+
+    def __init__(self):
+        super().__init__()
+        self._err_hashes = {}
+
+    def format(self, record):
+        if record.exc_info:
+            # Enable post mortem debugging
+            sys.last_type, sys.last_value, sys.last_traceback = record.exc_info
+            # Get default message, and its hash
+            msg = super().format(record)
+            msgh = hash(msg)
+            # If we've seen this message before, return a one-liner instead
+            if msgh in self._err_hashes:
+                count = self._err_hashes[msgh] + 1
+                self._err_hashes[msgh] = count
+                msg = record.funcName + "(): "
+                msg += str(sys.last_value).split("\n")[0].strip()
+                msg = msg if len(msg) <= 70 else msg[:69] + "…"
+                msg += f" ({count})"
+                return msg
+            else:
+                self._err_hashes[msgh] = 1
+                return msg
+        else:
+            return super().format(record)
+
+
+def create_logger():
+    logger = logging.getLogger("wgpu")
+    logger.propagate = False
+    logger.setLevel(logging.WARNING)
+
+    handler = logging.StreamHandler()
+    logger.addHandler(handler)
```

Comment:
https://docs.python.org/3/howto/logging.html#configuring-logging-for-a-library

In particular the note is relevant here

Reply:
Removed configuration of the logger. I realized that the storing of tb info on `sys.last_xx` is not needed if the `exc_info` argument is used. The shortening of recurring messages can be done where it was done before.